qdev_wrappers/show_num.py

```diff
@@ -4,46 +4,110 @@
 from qcodes.plots.qcmatplotlib import MatPlot
 
 
-def show_num(id, useQT=False, do_plots=True, **kwargs):
+def show_num(id, useQT=False, ave_col=False, ave_row=False, do_plot=True, savepng=False):
     """
     Show  and return plot and data for id in current instrument.
     Args:
         id(number): id of instrument
-        do_plots: Default False: if false no plots are produced.
-        **kwargs: Are passed to plot function
+        useQT (boolean): If true plots with QTplot instead of matplotlib
+        ave_col (boolean): If true subtracts average from each collumn
+        ave_row (boolean): If true subtracts average from each row
+        savepng (boolean): If true saves matplotlib figure as png
 
     Returns:
-        data, plots : returns the plot and the dataset
+        plot, data : returns the plot and the dataset
+
     """
-    if not getattr(CURRENT_EXPERIMENT, "init", True):
+    if not CURRENT_EXPERIMENT["init"]:
```

Comment:
I do not understand why we should consider the experiment as initialized, when `CURRENT_EXPERIMENT` does not have an entry for `init`

Reply:
This part was in show_num before i started modifying it. Not entirely sure what its purpose is.

Reply:
using `getattr` you can supply a default value (here `True`) if the key is not present in the dictionary. So I guess it is an improvement we should keep.